### DISCUSSION TITLE
Uncaught TypeError: (0 , import_node.createCookieSessionStorage) is not a function

### DISCUSSION THREAD

**Discussion by Fantros** [remix-run/remix#7666] [Q&A]
I'm getting this error `Uncaught TypeError: (0 , import_node.createCookieSessionStorage) is not a function` , can you help me?

```js
import { createCookieSessionStorage } from "@remix-run/node";

const { getSession, commitSession, destroySession } = createCookieSessionStorage({
	cookie: {
		name: "session",
		domain: ".example.com",
		secrets: ["test"],
		maxAge: 31536000
	}
});

export { getSession, commitSession, destroySession };
```

**Comment by Fantros** [comment]
I get this error when using useFetcher

**Comment by chatch** [comment]
also seeing this.  my workaround for now was to replace the import with require's and manual setup copied from the library:
```
const serverRuntime = require('@remix-run/server-runtime')
const crypto = require('@remix-run/node/dist/crypto')

const createCookie = serverRuntime.createCookieFactory({
  sign: crypto.sign,
  unsign: crypto.unsign,
})
const createCookieSessionStorage =
  serverRuntime.createCookieSessionStorageFactory(createCookie)
```

**Comment by rhatton77** [comment]
I also got this error when I move `createCookieSessionStorage` to its own file in `app/sessions.ts` like the [docs](https://remix.run/docs/en/main/utils/sessions#using-sessions) explains. 

The solution was to rename `app/sessions.ts` with the server extension to `app/sessions.server.ts` to ensure this only get complied on the server, as per [these docs](https://remix.run/docs/en/main/guides/constraints#no-module-side-effects)
